src/store/actions.ts

```diff
@@ -883,3 +639,433 @@ export const setStaffPayment: ActionIF = ({ commit }, staffPayment: StaffPayment
 export const setFolioNumber: ActionIF = ({ commit }, folioNumber: string): void => {
   commit('mutateFolioNumber', folioNumber)
 }
+/** Name Check actions
+ * TODO: move these into a factory if converting to composition api
+ */
+export const getMatchesExact = async (
+  { commit },
+  token: string,
+  cleanedName: string
+): Promise<Array<ConflictListItemI>> => {
+  const exactResp = await axios.get('/exact-match?query=' + cleanedName, {
+    headers: { Authorization: `Bearer ${token}`, 'Content-Type': 'application/json' }
+  }).catch(() => {
+    commit('mutateNameCheckErrorAdd', NameCheckErrorType.errorExact)
+    return null
+  })
+  return exactResp?.data ? parseExactNames(exactResp.data) : []
+}
+export const getMatchesSimilar = async (
+  { commit },
+  token: string,
+  cleanedName: string,
+  exactNames: Array<ConflictListItemI>
+): Promise<Array<ConflictListItemI>> => {
+  const synonymResp = await axios.get('/requests/synonymbucket/' + cleanedName + '/*', {
+    headers: { Authorization: `Bearer ${token}`, 'Content-Type': 'application/json' }
+  }).catch(() => {
+    commit('mutateNameCheckErrorAdd', NameCheckErrorType.errorSimilar)
+    return null
+  })
+  if (synonymResp?.data) synonymResp.data.exactNames = exactNames || []
+  return synonymResp?.data ? parseSynonymNames(synonymResp.data) : []
+}
+export const getMatchesRestricted = async (
+  { commit },
+  token: string,
+  cleanedName: string
+): Promise<ParsedRestrictedResponseIF> => {
+  const restrictedResp = await axios.get(`/documents:restricted_words?content=${cleanedName}`, {
+    headers: { Authorization: `Bearer ${token}`, 'Content-Type': 'application/json' }
+  }).catch(() => {
+    commit('mutateNameCheckErrorAdd', NameCheckErrorType.errorRestricted)
+    return null
+  })
+  return restrictedResp?.data
+    ? parseRestrictedWords(restrictedResp.data)
+    : { restrictedWords: [], conditionalWords: [] }
+}
+
+export const getNameAnalysis: ActionIF = async (
+  { commit, getters },
+  options: { xpro: boolean, designationOnly: boolean }
+) => {
+  const name = getters.getFullName
+  try {
+    if (options.designationOnly) commit('mutateAnalyzeDesignationPending', true)
+    else commit('mutateAnalyzeStructurePending', true)
+    commit('resetRequestExaminationOrProvideConsent')
+    const params: NewRequestNameSearchI = {
+      name: name,
+      location: getters.getLocation,
+      entity_type_cd: getters.getEntityTypeCd,
+      request_action_cd: getters.getRequestActionCd,
+      jurisdiction: options.xpro ? 'XPRO' : 'BC',
```

Comment:
Use enum values.

Reply:
Thanks. This makes is easier to understand what "type" this field is (as well as managing what values we can assign).

👍